src/main/scala/org/zalando/spearheads/innkeeper/services/team/TeamService.scala

```diff
@@ -1,23 +1,55 @@
 package org.zalando.spearheads.innkeeper.services.team
 
+import java.util.NoSuchElementException
 import com.google.inject.Inject
 import com.typesafe.config.Config
+import org.slf4j.LoggerFactory
 import org.zalando.spearheads.innkeeper.api.RouteOut
 import org.zalando.spearheads.innkeeper.oauth.AuthenticatedUser
+import org.zalando.spearheads.innkeeper.services.ServiceResult
+import org.zalando.spearheads.innkeeper.services.ServiceResult.{ Ex, NotFound, Result }
+import org.zalando.spearheads.innkeeper.services.team.TeamJsonProtocol._
+import org.zalando.spearheads.innkeeper.utils.HttpClient
+import scala.collection.immutable.Seq
+import scala.util.{ Try, Failure, Success }
 
 /**
  * @author dpersa
  */
 trait TeamService {
-  def hasSameTeamAsRoute(token: String, authenticatedUser: AuthenticatedUser, route: RouteOut): Boolean = true
 
-  def getForUsername(token: String, username: String): String = ""
+  def hasSameTeamAsRoute(authenticatedUser: AuthenticatedUser, route: RouteOut, token: String): Boolean = true
+
+  def getForUsername(username: String, token: String): Result[Team]
 }
 
-class ZalandoTeamService @Inject() (val config: Config) extends TeamService {
+class ZalandoTeamService @Inject() (val config: Config,
+                                    val httpClient: HttpClient) extends TeamService {
+
+  val logger = LoggerFactory.getLogger(this.getClass)
+
+  override def getForUsername(username: String, token: String): Result[Team] = {
+
+    (for {
+      json <- httpClient.callJson(url(username), Some(token))
+      teams <- Try { json.convertTo[Seq[Team]] }
+      officialTeam <- Try { teams.filter(_.teamType == Official).head }
```

Comment:
Part of the idea of mosaic is to make it as easy as possible to start new fragments and bring them live. I could imagine virtual teams (task forces?) to be spawned to rapidly prototype new ideas and bring them live as frictionless as possible... But I am not the one to make that call...